File: c++/215. Kth Largest Element in an Array.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        priority_queue<int> heap(nums.begin(),nums.end());
        int kthMax = 0;
        while(k--){
            if(k==0) kthMax = heap.top();
            heap.pop();
        }
        return kthMax;        
    }
};

File: leetcode/python/215. Kth Largest Element in an Array.py
import heapq
from typing import List

class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        heap = [-num for num in nums]
        heapq.heapify(heap)
        while k>0:
            popped = -heapq.heappop(heap)
            k-=1
        return popped

obj = Solution()
print(obj.findKthLargest([3,2,1,5,6,4],2))
print(obj.findKthLargest([3,2,3,1,2,4,5,5,6],4))